### PR TITLE
Removed file git history

### PR DESCRIPTION
**Pull request by tasdomas**
#### Summary
This pull request modifes the `file_unaltered` check in the sync tool, used to synchronize plugin repositories with the starter template, to handle the case where the target file does not exist.

### CODE REVIEW

File: a.txt [hunk 1]
```diff
@@ -0,0 +1 @@
+asdfadsf
```
Comment (hanzei) [inline a.txt:1]: Does this file need to be checked in?

File: build/sync/plan.yml [hunk 2]
```diff
@@ -42,3 +42,14 @@ actions:
         conditions:
           - type: exists
           - type: file_unaltered
+  - paths:
```
Comment (hanzei) [inline build/sync/plan.yml:45]: Same question here. Is this only for testing?

File: build/sync/plan/checks.go [hunk 3]
```diff
@@ -99,41 +99,67 @@ func (r PathExistsChecker) Check(path string, ctx Setup) error {
...
-	if info.IsDir() {
-		return fmt.Errorf("%q is a directory", absPath)
+
+	var srcDeleted bool
+	srcInfo, err := os.Stat(srcPath)
+	if os.IsNotExist(err) {
+		srcDeleted = true
+	} else if err != nil {
+		return fmt.Errorf("failed to get stat for %q: %v", trgPath, err)
+	}
```
Comment (hanzei) [inline build/sync/plan/checks.go:132]: Is this more readable? 

```suggestion
	if err != nil {
		if os.IsNotExist(err) {
			srcDeleted = true
		} else {
			return fmt.Errorf("failed to get stat for %q: %v", trgPath, err)
		}
	}
```

File: build/sync/plan/checks.go [hunk 4]
```diff
@@ -99,41 +99,67 @@ func (r PathExistsChecker) Check(path string, ctx Setup) error {
...
+			// Consider it unaltered.
+			return nil
+		}
+		// Check if the file was ever in git history.
+		_, err := git.FileHistory(path, setup.GetRepo(repo).Git)
+		if err == git.ErrNotFound {
+			// This is a new file being introduced to the target repo.
+			// Consider it unaltered.
+			return nil
+		} else if err != nil {
```
Comment (hanzei) [inline build/sync/plan/checks.go:150]: Is the `else` needed?

File: build/sync/plan/git/file_history.go [hunk 5]
```diff
@@ -22,6 +23,9 @@ func FileHistory(path string, repo *git.Repository) ([]string, error) {
 		FileName: &path,
 	}
 	commits, err := repo.Log(&logOpts)
+	if err == plumbing.ErrReferenceNotFound {
```
Comment (hanzei) [inline build/sync/plan/git/file_history.go:26]: Nit: Maybe use `errors.Is`?

File: build/sync/plan/checks.go [hunk 6]
```diff
@@ -99,41 +99,67 @@ func (r PathExistsChecker) Check(path string, ctx Setup) error {
...
+	trgInfo, err := os.Stat(trgPath)
+	if os.IsNotExist(err) {
+		if srcDeleted {
+			// File has been deleted in target and source repositories.
+			// Consider it unaltered.
+			return nil
+		}
+		// Check if the file was ever in git history.
+		_, err := git.FileHistory(path, setup.GetRepo(repo).Git)
+		if err == git.ErrNotFound {
```
Comment (hanzei) [inline build/sync/plan/checks.go:146]: Maybe use `errors.Is`?

File: go.mod [hunk 7]
```diff
@@ -3,7 +3,10 @@ module github.com/mattermost/mattermost-plugin-starter-template
 go 1.12
 
 require (
+	github.com/kr/text v0.2.0 // indirect
```
Comment (hanzei) [inline go.mod:6]: Should these changes be in `build/go.mod`?

File: build/sync/plan/checks_test.go [hunk 8]
```diff
@@ -8,8 +8,8 @@ import (
 	"path/filepath"
 	"testing"
 
+	git "github.com/go-git/go-git/v5"
```
Comment (hanzei) [inline build/sync/plan/checks_test.go:11]: Does `build/go.mod` need to get updated?

**Review by hanzei — CHANGES_REQUESTED** [review]
Thanks for fixing that issue :+1:

File: a.txt (hunk 1 above)
Comment (tasdomas) [reply a.txt:1]: No, no it doesn't. That's my mistake - I'll need to force push to resolve it though.

File: build/sync/plan.yml (hunk 2 above)
Comment (tasdomas) [reply build/sync/plan.yml:45]: Yes, this should not be here.

File: build/sync/plan/checks.go (hunk 3 above)
Comment (tasdomas) [reply build/sync/plan/checks.go:132]: I'm fine with both variants.

File: build/sync/plan/checks.go (hunk 4 above)
Comment (tasdomas) [reply build/sync/plan/checks.go:150]: Yes.

File: build/sync/plan/checks_test.go (hunk 8 above)
Comment (tasdomas) [reply build/sync/plan/checks_test.go:11]: I've updated go.mod.

File: build/go.sum [hunk 9]
```diff
@@ -113,7 +113,7 @@ github.com/couchbase/vellum v1.0.1/go.mod h1:FcwrEivFpNi24R3jLOs3n+fs5RnuQnQqCLB
 github.com/cpuguy83/go-md2man v1.0.10/go.mod h1:SmD6nW6nTyfqj6ABTjUi3V3JVMnlJmwcJI5acqYI6dE=
```
Comment (hanzei) [inline build/go.sum:113]: Can you please run `go mod tidy`?

**Review by hanzei — APPROVED** [review]
LGTM. One small request

File: go.mod (hunk 7 above)
Comment (mickmister) [reply go.mod:6]: @tasdomas What's your opinion on this?

File: build/go.sum (hunk 9 above)
Comment (tasdomas) [reply build/go.sum]: Done.

**Comment by jasonblais** [comment]
Great, thanks! @jfrerich ready for your review

**Review by jfrerich — APPROVED** [review]
Awesome work again, @tasdomas!

**Comment by hanzei** [comment]
Given that this is a tooling change and I have test it already locally, I'm going to skip QA review for this PR.

**Comment by hanzei** [comment]
Thanks @tasdomas for all the work you put into this tool :tada:

**Comment by hanzei** [comment]
The changes seem to have broken CI. @tasdomas Would you mind taking a look at https://app.circleci.com/pipelines/github/mattermost/mattermost-plugin-starter-template/428/workflows/508a6d30-001a-4087-b353-52dddc71683c/jobs/981?

**Comment by tasdomas** [comment]
@hanzei - the CI was broken by github squashing the PR's commits.

https://github.com/mattermost/mattermost-plugin-starter-template/pull/135 is a simple fix (just skipping one test)

**Pull request merged**